--- src-react/flattenChildren.js	2017-03-22 13:53:46.419278711 +0100
+++ src/flattenChildren.js	2017-03-22 14:33:15.957383210 +0100
@@ -12,9 +12,9 @@
 
 'use strict';
 
-var KeyEscapeUtils = require('KeyEscapeUtils');
-var traverseAllChildren = require('traverseAllChildren');
-var warning = require('warning');
+import KeyEscapeUtils from './KeyEscapeUtils';
+import traverseAllChildren from './traverseAllChildren';
+import warning from 'fbjs/lib/warning';
 
 var ReactComponentTreeHook;
 
@@ -99,4 +99,4 @@
   return result;
 }
 
-module.exports = flattenChildren;
+export default flattenChildren;
