--- src-react/ReactTransitionChildMapping.js	2017-03-22 11:50:17.107626479 +0100
+++ src/ReactTransitionChildMapping.js	2017-03-22 11:51:02.072450036 +0100
@@ -11,7 +11,7 @@
 
 'use strict';
 
-var flattenChildren = require('flattenChildren');
+import flattenChildren from './flattenChildren';
 
 var ReactTransitionChildMapping = {
   /**
@@ -102,4 +102,4 @@
   },
 };
 
-module.exports = ReactTransitionChildMapping;
+export default ReactTransitionChildMapping;
